Sources/ArcGISToolkit/Components/OfflineMapAreasView/OfflineMapAreasView.MapViewModel.swift

```diff
@@ -0,0 +1,168 @@
+// Copyright 2024 Esri
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//   https://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+import ArcGIS
+import Combine
+import Foundation
+
+extension OfflineMapAreasView {
+    /// The model class for the offline map areas view.
+    @MainActor
+    class MapViewModel: ObservableObject {
+        /// The portal item ID of the web map.
+        private var portalItemID: String?
+        
+        /// The offline map of the downloaded preplanned map area.
+        private var offlineMap: Map?
+        
+        /// The offline map task.
+        let offlineMapTask: OfflineMapTask
+        
+        /// The url for the preplanned map areas directory.
+        let preplannedDirectory: URL
+        
+        /// The mobile map packages created from mmpk files in the documents directory.
+        @Published private(set) var mobileMapPackages: [MobileMapPackage] = []
+        
+        /// The preplanned offline map information.
+        @Published private(set) var preplannedMapModels: Result<[PreplannedMapModel], Error>?
+        
+        /// A Boolean value indicating whether the map has preplanned map areas.
+        @Published private(set) var hasPreplannedMapAreas = false
+        
+        /// A Boolean value indicating whether the user has authorized notifications to be shown.
+        var canShowNotifications = false
+        
+        /// The job manager.
+        let jobManager = JobManager.shared
+        
+        init(map: Map) {
+            offlineMapTask = OfflineMapTask(onlineMap: map)
+            
+            if let portalItemID = map.item?.id?.rawValue {
+                FileManager.default.createDirectories(for: portalItemID)
+                
+                preplannedDirectory = FileManager.default.preplannedDirectory(portalItemID: portalItemID)
+            } else {
+                preplannedDirectory = FileManager.default.documentsDirectory
```

Comment:
it seems like this property is pulling double duty. Why is it that it has two different meanings?

Reply:
The property is now optional, the `else` case was a fallback for if the webmap id is `nil`.